docs/mobile-apps/automated-testing/about-appium.md

```diff
@@ -0,0 +1,437 @@
+
+```py
+driver.find_element_by_class_name('android.widget.DatePicker')
+```
+
+</TabItem>
+<TabItem value="id">
+
+In the mobile environment, `id`s are not, as in WebDriver, CSS ids, but rather some form of native identifier.
+
+* For iOS, the situation is complicated. Appium will first search for an `accessibility id` that matches. If none are found, a string match will be attempted on the element labels. Finally, if the `id` passed in is a localization key, it will search the localized string.
+* For Android, the `id` is the element’s `android:id`.
+
+This example below invokes the `findElement` method on the driver variable, using the name attribute to locate the `username` and `password` text input elements, and (optionally) the `id` attribute to locate the form element.
+
+__Example__:
+
+```java
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+WebElement emailInput = driver.findElement(By.id("fbemail"));
+```
+
+</TabItem>
+<TabItem value="xpath">
+
+The `xpath` locator strategy is also available in the WebDriver protocol, and exposes the functionality of [XPath language](https://www.w3.org/TR/xpath20/) to locate elements within a mobile view. An XML representation of the view is created in Appium, and searches are made against that image.
+
+The Selenium clients have methods for retrieving elements using the `xpath` locator strategy.
+
+__Example__:
+
+```py
+driver.find_element_by_xpath('//UIAApplication[1]/UIAWindow[1]/UIATextField[1]')
+```
+
+</TabItem>
+</Tabs>
+
+#### **Best Practices for Identifying Application Elements**
+
+We recommend using an element locator that uniquely identifies the element, like an `id` or an `accessibility id`. `class name` and `xpath` are best used only when IDs are not available.
+
+Multiple elements can have the same `class name`, and using `xpath` searches through the entire markup to find the element, which can slow down your tests.
+
+<br/>
+
+### 4. Perform Action(s) on the App Elements
+
+Once you've identified the mobile elements you want your test to interact with, the next step is to interact with them. You perform an action on a mobile element by invoking an interaction method on an instance of the WebElement interface.
+
+In this example, we'll test a login button. The WebElement interface declares basic interaction methods including:
+
+* The `sendKeys` method, to enter text.
+* The `clear` method, to clear entered text.
+* The `submit` method, to submit a form.
+
+This example first invokes the `sendKeys` method to enter text in the username and password elements, and then invokes the `submit` method to submit the login form. Enter a username and password:
+
+```java
+emailInput.sendKeys("SauceIsAwesome@email.com");
+```
+
+The `submit` method can be invoked either on any text input element on a form, or on the form element itself. Submit Text Element:
+
+```java
+emailInput.submit();
```

Comment:
You can't submit an input field in a native app, see also http://appium.io/docs/en/commands/element/other/submit/

Reply:
I didn't make this doc- simply ported over from old appium.md that is currently published here (https://docs.saucelabs.com/mobile-apps/automated-testing/appium)-- feel free to add any copy-editing commits you would like